internal/clients/http/sources/sources_client.go

```diff
@@ -222,9 +222,15 @@ func (c *sourcesClient) GetAuthentication(ctx context.Context, sourceId string)
 	if resp == nil {
 		return nil, fmt.Errorf("get source authentication call: empty response: %w", clients.ErrUnexpectedBackendResponse)
 	}
-	if resp.JSON404 != nil || resp.JSON400 != nil {
+	if resp.JSON404 != nil {
 		logger.Warn().Bytes("body", resp.Body).Int("status", resp.StatusCode()).Msg("Get authentication from sources returned 4xx")
+		return nil, fmt.Errorf("get source authentication call: %w", clients.ErrNotFound)
 	}
+	if resp.JSON400 != nil {
+		logger.Warn().Bytes("body", resp.Body).Int("status", resp.StatusCode()).Msg("Get authentication from sources returned 4xx")
+		return nil, fmt.Errorf("get source authentication call: %w", clients.ErrBadRequest)
+	}
+
```

Comment:
These two would be the most common cases - sources return 404, 400 or 2xx payloads. I think any other case could be a larger failure that we might want to know about anyway.
Therefore, after calling GetAuthentication, I feel like we want to check for ErrBadRequest and ErrNotFound, and the rest, like ErrNoResponseData, ErrUnexpectedBackendResponse should be left as is. :)